src/app/utils/assertions.ts

```diff
@@ -27,3 +27,15 @@ export const assertString = <X>(x: X, message: string): string => {
 
   return x;
 };
+
+// When the argument is "any", we don't want to cast it to object
+// or we'll be unable to assign it to more specific types
+// So we just exclude all primitive types instead
+export const assertObject = <X>(x: X, message: string) => {
+  // typeof null === 'object'
+  if (x === null || typeof x !== 'object') {
+    throw new Error(message);
+  }
+
+  return x as Exclude<X, number | string | boolean | bigint | symbol | null | undefined>;
```

Comment:
i have a different version of this https://github.com/openstax/ancillary-service/blob/main/utils/src/guards.ts#L6 i'm wondering if one or the other type conversion is better?

Reply:
I think yours is more strict, since it only allows non-class non-array non-function objects as far as I can tell. I use mine on instances of some class so I need the more lax type.